File: golang/lab4/Lab4.go
package lab4

import (
	"fmt"
	"math"
)

func Calculate(x float64) float64 {
	y := math.Sqrt(math.Pow(math.Asin(x), 4) + math.Pow(math.Acos(x), 4))
	return y
}

func TaskA(Xn, Xk, delX float64) []float64 {
	results := []float64{}
	for x := Xn; x <= Xk; x += delX {
		results = append(results, Calculate(x))
	}
	return results
}

func TaskB(x [5]float64) []float64 {
	results := []float64{}
	for _, value := range x {
		results = append(results, Calculate(value))
	}
	return results
}

func RunLab4() {
	fmt.Println(TaskA(0.22, 0.92, 0.14))
	var s = [5]float64{0.1, 0.35, 0.4, 0.55, 0.6}
	fmt.Println(TaskB(s))
}

File: python/src/lab4/lab4.py
import math


def Calculate(b, x: float) -> float:
    return (1 + (math.sin(b ** 3 + x ** 3) ** 2)) / ((b ** 3 + x ** 3) ** (1 / 3))


def TaskA(b, Xn, Xk, delX: float) -> list:
    arr = []
    while Xn <= Xk:
        arr.append(Calculate(b, Xn))
        Xn += delX
    return arr


def TaskB(b: float, x: list) -> list:
    arr = []
    for elem in x:
        arr.append(Calculate(b, elem))
    return arr


def PrintValue(Values):
    for elem in Values:
        print(elem)


def RunLab4Tasks():
    b = 2.5
    arr = [1.1, 2.4, 3.6, 1.7, 3.9]
    PrintValue(TaskA(b, 1.28, 3.28, 0.4))
    PrintValue(TaskB(b, arr))
